### PR TITLE
Added speakers list component to EventPage

### PR DESCRIPTION
**Pull request by oratiu**
Created a Speakers List component that gets included in the EventPage as per the requirements in this ticket:
https://asaphub.atlassian.net/jira/software/c/projects/CRN/boards/6?modal=detail&selectedIssue=CRN-a

### CODE REVIEW

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  ...gridMixin,
+  [`@media (max-width: ${mobileScreen.width}px)`]: {
+    gridAutoFlow: 'row',
+    alignItems: 'start',
+    borderBottom: '1px solid #DFE5EA',
+    [`&:nth-of-type(n+4)`]: {
+      display: 'none',
+    },
+  },
+  [`&:nth-of-type(n+6)`]: {
```
Comment (oratiu) [inline packages/react-components/src/organisms/SpeakersList.tsx:55]: Same here

File: packages/react-components/src/organisms/__tests__/SpeakerList.test.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { render } from '@testing-library/react';
+import { createEventResponse } from '@asap-hub/fixtures';
+import SpeakersList from '../SpeakersList';
+
+it('renders the speakers list', async () => {
+  const { getByText } = render(<SpeakersList {...createEventResponse()} />);
+
+  expect(getByText('The team one')).toBeVisible();
+  expect(getByText('To be announced')).toBeVisible();
+});
```
Comment (markcsims) [inline packages/react-components/src/organisms/__tests__/SpeakerList.test.tsx:10]: I couldn't see tests for the 'user was not announced' functionality

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      <div css={gridStyles}>
+        {speakers.map(({ user, team, role }) => (
+          <div key={team.id + user?.id} css={speakerListStyles}>
+            <div css={groupStyle}>
+              <div css={labelStyle}>
+                <span>Team</span>
+              </div>
+              <Link href={network({}).teams({}).team({ teamId: team.id }).$}>
+                {team?.displayName}
+              </Link>
```
Comment (markcsims) [inline packages/react-components/src/organisms/SpeakersList.tsx:120]: check link has the correct displayName and Id.

Here team has a ? at the end, I don't think this is necessary

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    <div>
+      <Headline3 styleAsHeading={3}>Speakers</Headline3>
+      <div css={headerStyle}>
+        <Headline4 styleAsHeading={4}>Team</Headline4>
+        <Headline4 styleAsHeading={4}>Speaker</Headline4>
+        <Headline4 styleAsHeading={4}>Role</Headline4>
+      </div>
+      <div css={gridStyles}>
+        {speakers.map(({ user, team, role }) => (
+          <div key={team.id + user?.id} css={speakerListStyles}>
```
Comment (markcsims) [inline packages/react-components/src/organisms/SpeakersList.tsx:113]: This isn't unique. What happens if a team has 2 unknown users.

Maybe use the index (and avoid the +)

File: packages/react-components/src/organisms/__tests__/SpeakerList.test.tsx (hunk 2 above)
Comment (markcsims) [reply packages/react-components/src/organisms/__tests__/SpeakerList.test.tsx:10]: We seem to be quite light on test coverage. 

The fixtures should be simplified to contain only the basics. We can then extend to test further with expectations and test descriptions explaining what we are testing.

**Review by markcsims — CHANGES_REQUESTED** [review]

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+import { EventSpeaker } from '@asap-hub/model';
+import { network } from '@asap-hub/routing';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Headline3, Headline4, Avatar, Link } from '../atoms';
+import { tabletScreen, perRem, mobileScreen } from '../pixels';
+import { userPlaceholderIcon } from '../icons';
+import { useDateHasPassed } from '../date';
+import { considerEndedAfter } from '../utils';
+
+const getPlaceholderAvatarUrl = () =>
```
Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:11]: I don't think you need to do this. Wont just {userPlaceholderIcon} work

File: packages/react-components/src/organisms/__tests__/SpeakerList.test.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+      const { getByText } = render(<SpeakersList {...event} />);
+
+      expect(getByText('To be announced')).toBeVisible();
+    });
+
+    it('Renders an announced user', async () => {
+      const event = {
+        ...createEventResponse(),
+        speakers: [{ ...announcedSpeaker }],
```
Comment (kmaid) [inline packages/react-components/src/organisms/__tests__/SpeakerList.test.tsx:42]: Don't think you need the spread

File: packages/react-components/src/templates/EventPage.tsx [hunk 7]
```diff
@@ -57,6 +59,7 @@ const EventPage: React.FC<EventPageProps> = ({
             {formatDistance(new Date(), new Date(lastModifiedDate))} ago
           </small>
         </Paragraph>
+        {props.speakers && <SpeakerList {...props} />}
```
Comment (kmaid) [inline packages/react-components/src/templates/EventPage.tsx:62]: What happens if speakers is an empty array? I think the SpeakerList should be omitted

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+                  imageUrl={
+                    user?.displayName
+                      ? user.avatarUrl || ''
+                      : getPlaceholderAvatarUrl()
+                  }
+                />
+                {(user?.displayName && (
+                  <Link
+                    href={
+                      user && network({}).users({}).user({ userId: user?.id }).$
```
Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:128]: Not sure why we have a ? in user.id and what happens for external users who don't have a profile

File: apps/storybook/src/SpeakersList.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+export default {
+  title: 'Organisms / Events / Speakers List',
+};
+
+const eventInThePast = {
+  ...createEventResponse(),
+  endDate: subHours(new Date(), 10).toISOString(),
+};
+
+export const Normal = () => <SpeakersList {...createEventResponse()} />;
```
Comment (kmaid) [inline apps/storybook/src/SpeakersList.stories.tsx:14]: Could this be more configurable so I can easily test different scenarios?

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    fontWeight: 'bold',
+  },
+  display: 'none',
+});
+
+const groupStyle = css({
+  display: 'flex',
+  flexFlow: 'column',
+  [`@media (max-width: ${mobileScreen.width}px)`]: {
+    paddingBottom: `${20 / perRem}em`,
```
Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:73]: This isn't divisible by 3

File: packages/react-components/src/templates/EventPage.tsx [hunk 11]
```diff
@@ -57,6 +59,7 @@ const EventPage: React.FC<EventPageProps> = ({
             {formatDistance(new Date(), new Date(lastModifiedDate))} ago
           </small>
         </Paragraph>
+        {props.speakers.length > 0 && <SpeakerList {...props} />}
```
Comment (kmaid) [inline packages/react-components/src/templates/EventPage.tsx:62]:
```suggestion
        {props.speakers.length && <SpeakerList {...props} />}
```

File: packages/validation/src/user-guards.ts [hunk 12]
```diff
@@ -1,5 +1,16 @@
...
+  ExternalAuthorResponse,
+  EventSpeakerUser,
+} from '@asap-hub/model';
 
 export const isInternalUser = (
   author: ExternalAuthorResponse | UserResponse,
 ): author is UserResponse => (author as UserResponse).email !== undefined;
+
+export const isInternalEventUser = (
+  user: EventSpeakerUser | undefined,
```
Comment (kmaid) [inline packages/validation/src/user-guards.ts:12]: Urgh, this is a weird type. I would have expected two types one InternalEventSpeaker one ExternalSpeaker.

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+            <div css={groupStyle}>
+              <div css={labelStyle}>
+                <span>Speaker</span>
+              </div>
+              <div css={userStyles}>
+                <Avatar
+                  {...user}
+                  imageUrl={
+                    isInternalEventUser(user)
+                      ? user?.avatarUrl || ''
```
Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:123]:
```suggestion
                      ? user?.avatarUrl 
```
Wouldn't this work?

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+                      : getPlaceholderAvatarUrl()
+                  }
+                />
+                {(isInternalEventUser(user) && (
+                  <Link
+                    href={
+                      user && network({}).users({}).user({ userId: user.id }).$
```
Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:130]:
```suggestion
				network({}).users({}).user({ userId: user.id }).$
```
?

Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:130]:
```suggestion
                      network({}).users({}).user({ userId: user.id }).$
```
?

File: packages/react-components/src/organisms/SpeakersList.tsx [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+                    }
+                  >
+                    {user?.displayName}
```
Comment (kmaid) [inline packages/react-components/src/organisms/SpeakersList.tsx:133]:
```suggestion
                    {user.displayName}
```

File: packages/validation/src/user-guards.ts (hunk 12 above)
Comment (kmaid) [reply packages/validation/src/user-guards.ts:12]: Actually i think External speakers are coming later

File: packages/validation/src/user-guards.ts [hunk 16]
```diff
@@ -1,5 +1,16 @@
...
+): boolean => {
```
Comment (kmaid) [inline packages/validation/src/user-guards.ts:13]:
```suggestion
):  user is EventSpeakerUser => {
```

File: packages/react-components/src/templates/EventPage.tsx [hunk 17]
```diff
@@ -57,6 +59,7 @@ const EventPage: React.FC<EventPageProps> = ({
             {formatDistance(new Date(), new Date(lastModifiedDate))} ago
           </small>
         </Paragraph>
+        {props.speakers.length && <SpeakerList {...props} />}
```
Comment (kmaid) [inline packages/react-components/src/templates/EventPage.tsx:62]:
```suggestion
        {!!props.speakers.length && <SpeakerList {...props} />}
```
JSX will print 0 without !!. Sorry

**Review by kmaid — APPROVED** [review]

**Review by markcsims — APPROVED** [review]

**Pull request merged**